drv/vsc7448/src/mac.rs

```diff
@@ -23,7 +23,20 @@ pub struct MacTableEntry {
     valid: bool,
 }
 
-pub fn next_mac(v: &impl Vsc7448Rw) -> Result<Option<MacTableEntry>, VscError> {
+pub fn count_macs(v: &impl Vsc7448Rw) -> Result<usize, VscError> {
+    // Reset MAC table reader, so FIND_SMALLEST starts from 0
```

Comment:
Possibly an ignorant question, not knowing the vsc7448 at all, but does the MAC table reader need to be reset in between calling `count_macs()` and iterating through with `next_mac()`?

Reply:
Good question!

Under the hood, we're doing a `FIND_SMALLEST` operation.  This operation reads the "current smallest" MAC address from `MAC_ACCESS_CFG0/1`, finds the next-smallest MAC address in the table, then writes it to those same registers.

This means if you do multiple `FIND_SMALLEST` calls, it will iterate over all MAC addresses in the table.  The final call will write the values back to 0.

This means that when `count_macs()` finishes, we've effectively reset the iterator, so there's no need to explicitly clear it again.

(If we have multiple tasks calling `next_mac`, then we'd want some kind of locking in place, but that's a problem for the future)

Reply:
Ahh, I see. Thanks!